proposals/0000-severe-warnings.rst

```diff
@@ -0,0 +1,130 @@
+
+* From the need to have *some* warnings being errors by default follows the need for the suitable infrastructure, hence the
+  ``-Wsevere`` warning group.
+  
+* Given that we have custom warning categories (`x-foo`, `#454 <https://github.com/ghc-proposals/ghc-proposals/pull/454>`_),
+  we anticipate that users of custom warning categories will want a way to add “their” warning group to ``-Wsevere`` as well.
+  
+  Because there is no registry of custom categories, we let the prefix indicate the group. The prefix ``x-`` indicates a custom warning in ``-Wdefault``.
+  This proposal allows the prefix ``xs-` to indicate that this group should be part of the severe group.
+   
+  
+Proposed Change Specification
+-----------------------------
+
+* GHC learns a new warning group flag ``-Wsevere``, next to the existing ``-Wdefault``, ``-Wextra``, `-Wall`` and ``-Wcompat``.
+  See _“5.2. Warnings and sanity-checking” <https://downloads.haskell.org/ghc/latest/docs/users_guide/using-warnings.html>_.
+  
+  It behaves like the other groups in that ``-Wno=severe``, ``-Wwarn=severe`` and ``-Werror=severe`` behave as if the flag
+  had been specified for each warning in the groups.
+  
+* The warning group is initialized with the following warning categories:
+
+  * ``missing-methods``
+  * ``missing-fields``
+
+*  The set of allowed names of custom warning categories (`#454 <https://github.com/ghc-proposals/ghc-proposals/pull/454>`_) is
+   extended by those prefixed with ``xs-*` (for “custom extended – severe”). The flags
+   ``-Wno=severe``, ``-Wwarn=severe`` and ``-Werror=severe``
+   also affect all custom warnigns with ah name starting in ``xs-*``.
+ 
+
+Examples
+--------
+TODO
+
+
+Effect and Interactions
+-----------------------
+TODO
+
+Costs and Drawbacks
+-------------------
+In terms of GHC development, this is a modest extension of the existing warning category and group infrastructure.
+
+In terms of breakage, @phadej has made an impact analysis at <https://github.com/ghc-proposals/ghc-proposals/issues/544#issue-1410125536>.
+
+
+Alternatives
+------------
+
+* Naming the group.
+
+  I suggest to use the name ``severe``, which does not have ``error`` in it (e.g. ``errors-by-default``), because else
+  ``-Werror=errors-by-default`` or something looks kinda strange, and after someone says ``-Wwarn=severe``, the ``severe`` group still
+  exists, but it not an error.
+  
+  I briefly considered ``-Wfatal`, but that’s a lie – these errors are *not* ``fatal``, else we couldn’t turn them
+  into warnings.
+  
+* We could add more warnings to the group right away.
+
+* In particular, we could make ``partial-type-signatures`` a normal warning, in ``-Wsevere`` by default, and ``-XPartialTypeSignatures``
+  becomes a synonym for ``-Wwarn=partial-type-signatures``.
+  
+  (Or should ``-XPartialTypeSignatures`` be a flag that changes the *warning group* of ``partial-type-signatures``? Not sure).
```

Comment:
I think some form of this is probably a good idea, but I suggest to keep the current proposal simple and deal with this in a follow up proposal at some point, once the basic `-Wsevere` group is established.